ingenialink/canopen/network.py

```diff
@@ -354,6 +362,8 @@ def _setup_connection(self) -> None:
         """Creates a network interface object establishing an empty connection
         with all the network attributes already specified."""
         if self._connection is None:
+            if not RUNNING_ON_WINDOWS and self.__device != CAN_DEVICE.SOCKETCAN.value:
```

Comment:
I would move this check to the constructor.

Reply:
We should check if the OS is linux.

Reply:
Done [9d6f708](https://github.com/ingeniamc/ingenialink-python/pull/390/commits/9d6f708707fbcf386670c5534ef19ed948f7e0bb)